File: tests/fixtures/python/pipeline.py
"""Entry point and configuration for the processing pipeline."""

from .processor import DataProcessor, JSONProcessor, create_default_processor
from .filters import NormalizeFilter, chain_filters


class PipelineConfig:
    def __init__(self, mode: str = "default"):
        self.mode = mode

    def build_pipeline(self):
        if self.mode == "default":
            return create_default_processor()
        proc = JSONProcessor(self.as_dict())
        flt = chain_filters(NormalizeFilter())
        proc.add_transformer(flt)
        return proc

    def as_dict(self) -> dict:
        return {"mode": self.mode, "version": 1}


def run_pipeline(input_data: list[dict]) -> list[dict]:
    config = PipelineConfig()
    proc = config.build_pipeline()
    return [proc.process(d) for d in input_data]

File: tests/fixtures/typescript/pipeline.ts
// Pipeline.ts — entry point, cross-module composition

import {
  DataProcessor,
  JSONProcessor,
  createDefaultProcessor,
  ProcessorConfig,
} from "./processor";
import { NormalizeFilter, chainFilters } from "./filters";

export class PipelineConfig {
  private mode: string;

  constructor(mode: string = "default") {
    this.mode = mode;
  }

  buildPipeline(): DataProcessor {
    if (this.mode === "default") {
      return createDefaultProcessor();
    }
    const proc = new JSONProcessor(this.toConfig());
    proc.addTransformer(chainFilters(new NormalizeFilter()));
    return proc;
  }

  private toConfig(): ProcessorConfig {
    return { mode: this.mode, version: 1 } as ProcessorConfig;
  }
}

export function runPipeline(
  inputData: Record<string, unknown>[]
): Record<string, unknown>[] {
  const config = new PipelineConfig();
  const proc = config.buildPipeline();
  return inputData.map((d) => proc.process(d));
}
